src/main/kotlin/grpc/ValidationInterceptor.kt

```diff
@@ -0,0 +1,84 @@
+package se.uulm.snowballr.backend.grpc
+
+import arrow.core.Either
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.grpc.ForwardingServerCallListener.SimpleForwardingServerCallListener
+import io.grpc.Metadata
+import io.grpc.ServerCall
+import io.grpc.ServerCallHandler
+import io.grpc.ServerInterceptor
+import io.grpc.Status
+import se.uulm.snowballr.backend.validation.validateRequest
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * A server interceptor responsible for validating incoming requests in gRPC communication.
+ *
+ * This interceptor wraps the original server call listener using a custom [ValidationCallListener],
+ * which performs input validation on received messages. If validation fails, the listener closes
+ * the call with an [Status.INVALID_ARGUMENT] status and a detailed error description.
+ */
+val validationInterceptor =
+    object : ServerInterceptor {
+        override fun <ReqT : Any?, RespT : Any?> interceptCall(
+            call: ServerCall<ReqT?, RespT?>?,
+            headers: Metadata?,
+            next: ServerCallHandler<ReqT?, RespT?>?,
+        ): ServerCall.Listener<ReqT?>? {
```

Comment:
```suggestion
        ): ServerCall.Listener<ReqT?> {
```

My linter tells me, that this call always returns a non-null type.

Reply:
The method of `ServerInterceptor` is a java method, which can always return null.
As we do not return a nullable value, we can omit the `?`.